### ISSUE TITLE
Ungraceful error when filtering on invalid job status in the runs page

### ISSUE THREAD

**Issue by rexledesma** [dagster-io/dagster#7219]
## Summary
Running into 

```
Operation name: RunsRootQuery

Message: Variable "$filter" got invalid value {"statuses": ["sd"]}.
In field "statuses": In element #0: Expected type "RunStatus", found "sd".

Path: 

Locations: [{"line":1,"column":51}]
```

We should have a client-side validation of the run status input before initiating a GraphQL request.

## Reproduction
1. Navigate to `/instance/runs`.
2. Filter by `status:` and ignore autocomplete, e.g. use `status:asd`
3. See GraphQL error.

## Dagit UI/UX Issue Screenshots
![Screen Shot 2022-03-28 at 2 54 28 PM](https://user-images.githubusercontent.com/16431325/160493663-bd1bbc92-8536-4c75-bfab-fa2977a9265a.png)

## Additional Info about Your Environment
Dagster 0.14.6

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by hellendag** [comment]
Yeah, it's not great to surface user errors like these as "GraphQL errors". I'll see if I can come up with something more elegant.
